src/main/java/com/github/princesslana/greedorama/commands/HistoryCommand.java

```diff
@@ -0,0 +1,92 @@
+package com.github.princesslana.greedorama.commands;
+
+import com.github.princesslana.greedorama.PortfolioRepository;
+import com.github.princesslana.greedorama.Transaction;
+import com.github.princesslana.greedorama.UserRepository;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Ordering;
+import disparse.discord.smalld.DiscordRequest;
+import disparse.discord.smalld.DiscordResponse;
+import disparse.parser.reflection.CommandHandler;
+import disparse.parser.reflection.Flag;
+import disparse.parser.reflection.ParsedEntity;
+import java.util.HashSet;
+
+public class HistoryCommand {
+
+  private static final int TXNS_PER_PAGE = 10;
+
+  private final DiscordRequest request;
+
+  private final PortfolioRepository portfolios;
+
+  private final UserRepository users;
+
+  public HistoryCommand(
+      DiscordRequest request, PortfolioRepository portfolios, UserRepository users) {
+    this.request = request;
+    this.portfolios = portfolios;
+    this.users = users;
+  }
+
+  @ParsedEntity
+  private static class Options {
+    @Flag(shortName = 'p', longName = "page", description = "Display page")
+    public Integer page = 1;
+  }
+
+  @CommandHandler(commandName = "history")
+  public DiscordResponse portfolio(Options options) {
+    var guildId = request.getDispatcher().guildFromEvent(request.getEvent());
+    var userId = request.getDispatcher().identityFromEvent(request.getEvent());
+
+    var user = users.get(guildId, userId);
+    var portfolio = portfolios.get(user);
+
+    var txns = portfolio.getTransactions();
+    var totalPages = (txns.size() + TXNS_PER_PAGE - 1) / TXNS_PER_PAGE;
+    var page = options.page;
+
+    var pageText = String.format("Page %d of %d", options.page, totalPages);
+
+    if (options.page > totalPages || options.page <= 0) {
+      return Format.error("Page number out of range");
+    }
+
+    var dates = new HashSet<String>();
+    var txnList = new StringBuilder();
+
+    var byWhen = Ordering.natural().reverse().onResultOf(Transaction::getWhen);
+    txns = byWhen.sortedCopy(txns);
+
+    for (var i = (page - 1) * TXNS_PER_PAGE; i < page * TXNS_PER_PAGE; i++) {
+
+      if (i >= txns.size()) {
+        break;
+      }
+
+      var txn = txns.get(i);
+      var date = Format.date(txn.getWhen());
+
+      var line =
+          String.format(
```

Comment:
Do you have an example of what this output produces?

Reply:
![HistoryExample](https://user-images.githubusercontent.com/41343736/86956168-8a76d280-c10d-11ea-878e-dca11c8189c9.JPG)

Each individual transaction line is one iteration of the for loop.

I store the dates (day/month/year) in a set since the output seems a bit too cluttered when the same date is repeated constantly, as shown below, although I could get rid of that if needed.

![HistoryExample2](https://user-images.githubusercontent.com/41343736/86956332-c1e57f00-c10d-11ea-80d9-bf6af06c806b.JPG)

Reply:
I think we can remove the time. I may come back and change the layout to be more table like, kind of like portfolio, but I'm not entirely sure how I'd want it yet, lol.

I do like the the way you've done the date only shown once rather than every line, that's cool.

Reply:
Thanks!